### PR TITLE
editoast: Utoipa for scenarios

### PR DESCRIPTION
**Pull request by Tristramg**

### CODE REVIEW

File: editoast/src/models/scenario.rs [hunk 1]
```diff
@@ -174,9 +187,15 @@ impl List<(i64, Ordering)> for ScenarioWithCountTrains {
     ) -> Result<PaginatedResponse<Self>> {
         let study_id = params.0;
         let ordering = params.1.to_sql();
-        sql_query(format!("SELECT t.*,COUNT(train_schedule.id) as trains_count FROM scenario as t
+        sql_query(format!("WITH scenarios_with_train_counts AS (
```
Comment (leovalais) [inline editoast/src/models/scenario.rs:190]: I'm fine with the change but why do you want to add the infra name?

File: editoast/src/views/scenario.rs [hunk 2]
```diff
@@ -172,6 +212,16 @@ impl From<ScenarioPatchForm> for Scenario {
     }
 }
 
+/// Update a scenario
+#[utoipa::path(
+    tag = "scenarios",
+    params(ProjectIdParam, StudyIdParam, ScenarioIdParam),
+    request_body = ScenarioPatchForm,
+    responses(
+        (status = 204, description = "The scenario was deleted successfully"),
```
Comment (leovalais) [inline editoast/src/views/scenario.rs]:
```suggestion
        (status = 204, description = "The scenario was updated successfully"),
```

**Review by leovalais — APPROVED** [review]
Thanks! Tested using the frontend.

File: editoast/src/models/scenario.rs (hunk 1 above)
Comment (Tristramg) [reply editoast/src/models/scenario.rs:190]: It is required in the front. I’m not sure how it worked before. It is already returned in `ScenarioWithDetails`

**Review by clarani — APPROVED** [review]
LGTM, tested ! :white_check_mark:

**Pull request merged**